pkg/discovery/discovery.go

```diff
@@ -39,6 +40,21 @@ func NamespacedGVRs(ctx context.Context, cli discovery.DiscoveryInterface) ([]sc
 	return apiToGroupVersion(arls)
 }
 
+func NamespacedGVRsIgnoreAPIGroupErr(ctx context.Context, cli discovery.DiscoveryInterface, exclude filter.ResourceTypeMatcher) ([]schema.GroupVersionResource, error) {
+	arls, err := cli.ServerPreferredNamespacedResources()
+	if err == nil {
+		return apiToGroupVersion(arls)
+	}
+	out, _ := err.(*discovery.ErrGroupDiscoveryFailed)
+	for k, _ := range out.Groups {
```

Comment:
`out.Groups` may be nil if the type cast failed. Please check the typecast and if err isn't an `ErrGroupDiscoveryFailed`, then return it from this function

Reply:
ok. Planning to use `discovery.IsGroupDiscoveryFailedError(err)` check

Reply:
you already are doing the typecast to get out.Groups - we can just use that